src/shallot.py

```diff
@@ -26,6 +29,9 @@ def send_tcp(ip: str, port: int, data: bytes):
     if ip == list_server.my_public_ip:
         ip = 'localhost'
     # TODO open a TCP socket to the (IP, port), send the data, and close the connection
+    with socketserver.socket(socketserver.AF_INET, socketserver.SOCK_STREAM) as sock:
```

Comment:
I don't think socketserver has AF_INET or SOCKSTREAM ? It's a socket library that has it. Did you test it? If so I would like to know how.

Even if I modify your code to correct purpose, it really hard to test the whole code base in general. The `make_request` function is not working well. Essentially after `send_tcp`, the handle() is not being called from `ShallotHandler` class.

Reply:
Yeah this was an autocomplete typo. I fixed the issue and pushed code.